wombat/src/main/cpp/utils/Encoder.cpp

```diff
@@ -45,88 +43,93 @@ units::radian_t Encoder::GetEncoderPosition() {
-    controller->ConfigSelectedFeedbackSensor(ctre::phoenix::motorcontrol::TalonFXFeedbackDevice::IntegratedSensor);
-  }
+utils::TalonFXEncoder::TalonFXEncoder(
+    ctre::phoenix::motorcontrol::can::TalonFX *controller, double reduction)
+    : utils::Encoder(2048, reduction, 0), _controller(controller) {
+  controller->ConfigSelectedFeedbackSensor(
+      ctre::phoenix::motorcontrol::TalonFXFeedbackDevice::IntegratedSensor);
+}
 
-double TalonFXEncoder::GetEncoderRawTicks() const {
+double utils::TalonFXEncoder::GetEncoderRawTicks() const {
   return _controller->GetSelectedSensorPosition();
 }
 
-double TalonFXEncoder::GetEncoderTickVelocity() const {
+double utils::TalonFXEncoder::GetEncoderTickVelocity() const {
   return _controller->GetSelectedSensorVelocity() * 10;
 }
 
+utils::TalonSRXEncoder::TalonSRXEncoder(
+    ctre::phoenix::motorcontrol::can::TalonSRX *controller,
+    double ticksPerRotation, double reduction)
+    : utils::Encoder(ticksPerRotation, reduction, 0), _controller(controller) {
+  controller->ConfigSelectedFeedbackSensor(
+      ctre::phoenix::motorcontrol::TalonSRXFeedbackDevice::QuadEncoder);
+}
 
-TalonSRXEncoder::TalonSRXEncoder(ctre::phoenix::motorcontrol::can::TalonSRX *controller, double ticksPerRotation, double reduction) 
-  : Encoder(ticksPerRotation, reduction, 0), _controller(controller) {
-    controller->ConfigSelectedFeedbackSensor(ctre::phoenix::motorcontrol::TalonSRXFeedbackDevice::QuadEncoder);
-  }
-
-double TalonSRXEncoder::GetEncoderRawTicks() const {
+double utils::TalonSRXEncoder::GetEncoderRawTicks() const {
   return _controller->GetSelectedSensorPosition();
 }
 
-double TalonSRXEncoder::GetEncoderTickVelocity() const {
+double utils::TalonSRXEncoder::GetEncoderTickVelocity() const {
   return _controller->GetSelectedSensorVelocity() * 10;
 }
 
+utils::DutyCycleEncoder::DutyCycleEncoder(int channel, double ticksPerRotation,
+                                          double reduction)
+    : utils::Encoder(ticksPerRotation, reduction, 0),
+      _dutyCycleEncoder(channel) {}
 
-
-DutyCycleEncoder::DutyCycleEncoder(int channel, double ticksPerRotation, double reduction) 
-  : Encoder(ticksPerRotation, reduction, 0), _dutyCycleEncoder(channel) {}
-
-double DutyCycleEncoder::GetEncoderRawTicks() const {
+double utils::DutyCycleEncoder::GetEncoderRawTicks() const {
   return _dutyCycleEncoder.Get().value();
 }
 
-double DutyCycleEncoder::GetEncoderTickVelocity() const {
-  return 0;
-}
+double utils::DutyCycleEncoder::GetEncoderTickVelocity() const { return 0; }
 
-CanEncoder::CanEncoder(int deviceNumber, double ticksPerRotation, double reduction)
-  : Encoder(ticksPerRotation, reduction, 1) {
-    _canEncoder = new CANCoder(deviceNumber, "Drivebase");
-  }
+utils::CanEncoder::CanEncoder(int deviceNumber, double ticksPerRotation,
+                              double reduction)
+    : utils::Encoder(ticksPerRotation, reduction, 1) {
+  _canEncoder = new CANCoder(deviceNumber, "Drivebase");
```

Comment:
the can bus name should probably be changeable

Reply:
Good idea fr fr